src/nbl/video/utilities/IUtilities.cpp

```diff
@@ -192,458 +96,48 @@ void IUtilities::updateImageViaStagingBuffer(
         }
         else
         {
-            // Start CmdCopying Regions and Copying Data to m_defaultUploadBuffer
-            uint32_t currentUploadBufferOffset = 0u;
-            uint32_t availableUploadBufferMemory = 0u;
-            auto addToCurrentUploadBufferOffset = [&](uint32_t size) -> bool 
-            {
-                currentUploadBufferOffset += size;
-                currentUploadBufferOffset = core::alignUp(currentUploadBufferOffset, bufferOffsetAlignment);
-                if(currentUploadBufferOffset - localOffset <= uploadBufferSize)
-                {
-                    availableUploadBufferMemory = uploadBufferSize - (currentUploadBufferOffset - localOffset);
-                    return true;
-                }
-                else
-                    return false;
-            };
-
-            // currentUploadBufferOffset = localOffset
-            // currentUploadBufferOffset = alignUp(currentUploadBufferOffset, bufferOffsetAlignment)
-            // availableUploadBufferMemory = uploadBufferSize
-            addToCurrentUploadBufferOffset(localOffset);
+            size_t currentUploadBufferOffset = localOffset;
+            size_t availableUploadBufferMemory = allocationSize;
 
             bool anyTransferRecorded = false;
```

Comment:
redundant variable, can just check `!regionsToCopy.empty()`

Reply:
You know, when you evolve from an old code to a newer version, you don't question what already works 🤣